pkg/autoscaler/provider.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2019 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package autoscaler
```

Comment:
Am I correct that with this we should be able to add e2e testing for HPA-class autoscaling based on request rate?  I'm constantly worried about that codepath rotting (and the same for this), so the sooner we can get e2e testing in place for this whole thing e2e, the happier I will be :)

Reply:
Yes, that‘s the ultimate goal that will conclude this journey. /metric=concurrency for the HPA and then duplicating the AutoscaleTests for that.